src/routing/Page.tsx

```diff
@@ -14,8 +15,11 @@ export default function Page(props: PageProps): JSX.Element {
     console.log('submiting');
   };
 
+  const { path } = useRouteMatch();
+  const combinedPath = buildRelativePath(path, props.path);
```

Comment:
I think another way we could approach this is to check the length of `path`, and if it's greater than 1, build the route path using a template string. This way, we wouldn't necessarily have to rely on stack overflow code:

```
<Route path={path.length > 1 ? `${path}${props.path}` : `${props.path}`}>
```